backend/app/Repositories/UserRepository.php

```diff
@@ -0,0 +1,18 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Repositories;
+
+use App\User;
+use App\Repositories\Contracts\UserRepository as EloquentUserRepository;
+
+final class UserRepository implements EloquentUserRepository
```

Comment:
not exactly, your interface was called correctly, because interface is general and does not depend on implementation. So when you use UserRepository you are not worried about where the user has been stored. But it is important when you binding implementaion to interface, because some repository can store your data via Eloquent framework, or in the files, or in the memory.

So rename class to EloquentUserRepository and interface leave as it was